packages/server/utils/OpenAIServerManager.ts

```diff
@@ -40,6 +41,86 @@ class OpenAIServerManager {
       return null
     }
   }
+
+  async groupReflections(reflectionsText: string[]) {
+    if (!this.openAIApi) return null
+    try {
+      const response = await this.openAIApi.createChatCompletion(
+        {
+          model: 'gpt-3.5-turbo',
```

Comment:
I tried testing this with the reflections in our previous Squad's retro, but it really struggles to group reflections when they're not similar. This often leads to a few groups being created with many reflections but several with just one reflection.

I've tried many different iterations of saying: "Never create groups with just one reflection", but it struggles with this.

I have tried it with the gpt-4 model, and it works significantly better. However, the token limit is too low. Also, the gpt-3.5-turbo model can take 10-20 seconds to respond, so the gpt-4 model can hit the 60 seconds timeout limit.

We could group all of the solo groups together into one "Other" group. I suspect it'll be hard to launch the feature to all users until gpt-4 comes out of beta, which I would imagine will happen within a month or two, given the rate of recent progress. 

![Screenshot 2023-05-19 at 16 34 06](https://github.com/ParabolInc/parabol/assets/39854876/437ac39b-225a-4e94-b9a3-0d7292da2c92)